Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -495,6 +496,16 @@ private static string FindCommandToRun(FileInfoBase[] files, out IScriptHost scr
                         }
                     }
                 }
+
+                foreach (var supportedFileName in scriptHost.SupportedFileNames)
```

Comment:
Should the SupportedFileNames check be performed before supported extensions? For example, what happens if there are .js files in a Dnx directory root - would the Node ScriptHost be selected?